models/transfers/optimism/transfers_optimism_tokens.sql

```diff
@@ -0,0 +1,159 @@
+{{ 
+    config(
+        alias ='tokens', 
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_transfer_id',
+        post_hook='{{ expose_spells(\'["optimism"]\',
+                                    "sector",
+                                    "transfers",
+                                    \'["msilb7", "chuxin"]\') }}'
+    )
+}}
+
+SELECT transfer_from_address,
+        transfer_to_address,
+        contract_address,
+        token_standard,
+        token_type,
+        token_id,
+        transfer_type,
+        -- mint, burn, or transfer
+        CASE
+                WHEN transfer_from_address = '0x0000000000000000000000000000000000000000' THEN 'mint'
+                WHEN transfer_to_address = '0x0000000000000000000000000000000000000000' THEN 'burn'
+                ELSE 'transfer'
+                END 
+        AS transfer_action,
+
+        -- is the transfer part of the top-level transaction, or is it internal?
+        CASE WHEN substring(t.data, 1, 10)
+                        IN (    '0x42842e0e','0x23b872dd','0xb88d4fde','0xf3993d11' --erc721
+                                ,'0xf242432a','0x2eb2c2d6' --erc1155
+                                , '0xa9059cbb' --erc20
+                        )
+                        THEN 'transfer transaction'
+                WHEN t.gas_used = 21000 AND substring(t.data, 1, 10) = '0x'
+                        THEN 'transfer transaction' -- we specify gas_used because some arb bots will send null data as well.
+                WHEN substring(t.data, 1, 10) = '0xd0e30db0'
+                        THEN 'eth wrap'
+                WHEN substring(t.data, 1, 10) = '0x2e1a7d4d'
+                        THEN 'eth unwrap'
+                ELSE 'internal transfer'
+                END
+        AS transfer_location,
+
+        cast(tfs.value as double) AS value,
+        tx_block_time,
+        DATE_TRUNC('day',tx_block_time) AS tx_block_date,
+        tx_block_number,
+        tx_hash,
+        substring(t.data, 1, 10) AS tx_method_id,
+        t.from AS tx_from_address,
+        t.to AS tx_to_address,
+        evt_index,
+        trace_address,
+        unique_transfer_id
+
+FROM (
+        -----------
+        -- ERC20 --
+        -----------
+        SELECT
+
+        r.from AS transfer_from_address,
+        r.to AS transfer_to_address,
+        r.contract_address AS contract_address,
+        'erc20' AS token_standard,
+        'fungible' AS token_type,
+        '' AS token_id, -- used by NFTs
+        '' AS transfer_type, -- used by NFTs
```

Comment:
Also this part for `transfer_type` shoule be 'single' or 'batch' ?
If none of these we may need to add new accepted values in schema.

Reply:
I'll remove the test. This should be handled for in the NFT tables anyway.